MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -54,20 +54,21 @@ struct QuestionFactory: QuestionFactoryProtocol {
         self.delegate = delegate
     }
     
-    mutating func shuffleQuesions() {
+    func shuffleQuesions() {
         self.questionShuffled = questions?.shuffled()
     }
     
-    mutating func requestNextQuestion() {
+    func requestNextQuestion() {
         
-        guard let _ = questions else {
+        guard let _ = questionShuffled else {
```

Comment:
Если написать так:
```suggestion
        guard let questionShuffled = questionShuffled else {
```

то ниже можно использовать так:
delegate?.didReceiveNextQuestion(question: questionShuffled[safe: 0])

иначе guard использован как-то очень странно :)